src/main/java/com/linecorp/armeria/server/docs/ThriftDocString.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.stream.XMLStreamReader;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * {@link ThriftDocString} is an DocString extractor for Thrift IDL JSON.
+ * If you want use DocString in {@link DocService}, generate YOUR.json using YOUR.thrift and
+ * latest Thrift compiler with --gen json option. And, place YOUR.json file to
+ * resources/&#42;&#42;&#47;thriftjson&#47;&#42;&#42;&#47;YOUR.json of package paths.
+ */
+public final class ThriftDocString {
+
+    private static final Pattern FILENAME_FILTER = Pattern.compile(".*json");
+
+    private static final Pattern FULLPATH_FILTER = Pattern.compile(".*/thriftjson/.*\\.json");
```

Comment:
I think it's more robust and less thinking for users if we just define the absolute location ourselves, the extra flexibility doesn't seem to offer much.

com.linecorp.armeria.server.docs.thriftjson seems ok to me